### PR TITLE
switch to using @manypkg/get-packages for enumerating monorepo packages

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

This switches out usage of `@lerna` packages in `backend-common` and `cli` with `@manypkg/get-packages` and custom implementations. It removes our direct dependency on lerna and allows other types of monorepo setups to be used.

This should also remove some ReDoS vulnerabilities from the updated packages, although nothing that can be exploited in practice

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/cli/src/lib/monorepo/PackageGraph.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+import { Package } from '@manypkg/get-packages';
+
+type PackageJSON = Package['packageJson'];
+
+export interface ExtendedPackageJSON extends PackageJSON {
+  scripts?: {
+    [key: string]: string;
+  };
+  // The `bundled` field is a field known within Backstage, it means
```
Comment (freben) [inline packages/cli/src/lib/monorepo/PackageGraph.ts:25]: Sounds like a candidate for lifting into what comes out of #8729

Comment (freben) [reply packages/cli/src/lib/monorepo/PackageGraph.ts:25]: At which point a `BackstagePackageJsonExtensions` type in `@backstage/types` might be interesting

**Review by freben — APPROVED** [review]
Ahh that is indeed nicer too. Great job!

File: packages/cli/src/lib/monorepo/PackageGraph.ts (hunk 1 above)
Comment (Rugvip) [reply packages/cli/src/lib/monorepo/PackageGraph.ts:25]: Yep idd. Perhaps not `@backstage/types` though as it's really only interesting for tooling, probably `@backstage/cli-common`

Comment (freben) [reply packages/cli/src/lib/monorepo/PackageGraph.ts:25]: Ah yep, makes sense

**Pull request merged**